File: src/test/java/com/lffblk/counter/TestIPGenerator.java
package com.lffblk.counter;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

class TestIPGenerator {

    private final Random random = new Random();

    List<String> generateIpAddressOctets() {
        return generateIpAddressOctetsInternal();
    }

    String generateIpAddress() {
        List<String> octets = generateIpAddressOctetsInternal();
        return String.join(".", octets);
    }

    private List<String> generateIpAddressOctetsInternal() {
        List<String> octets = new ArrayList<>(4);
        for (int i = 0; i < 4; i++) {
            octets.add(String.valueOf(random.nextInt(256)));
        }
        return octets;
    }
}

File: src/test/java/com/lffblk/counter/TestIPGenerator.kt
package com.lffblk.counter

import java.util.ArrayList
import java.util.Random

internal class TestIPGenerator {

    private val random = Random()

    fun generateIpAddressOctets(): List<String> {
        return generateIpAddressOctetsInternal()
    }

    fun generateIpAddress(): String {
        val octets = generateIpAddressOctetsInternal()
        return octets.joinToString(".")
    }

    private fun generateIpAddressOctetsInternal(): List<String> {
        val octets = ArrayList<String>(4)
        for (i in 0..3) {
            octets.add(random.nextInt(256).toString())
        }
        return octets
    }
}
